addons/survey/models/survey_user_input.py

```diff
@@ -269,6 +273,23 @@ def get_print_url(self):
         self.ensure_one()
         return '%s?answer_token=%s' % (self.survey_id.get_print_url(), self.access_token)
 
+    def _get_lang_with_fallback(self):
+        """ :return: the most suitable language for the user that is supported by the survey. """
+        self.ensure_one()
+        lang_code = self.lang_id.code or self.env.context.get('lang') or self.env['ir.http']._get_default_lang().code
```

Comment:
As this only work with a `request`, shouldn't it be a controller method/util instead?